dialogue-annotations/src/main/java/com/palantir/dialogue/annotations/HttpEntityBodyRequestBodyAdapter.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * (c) Copyright 2021 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.dialogue.annotations;
+
+import com.palantir.dialogue.RequestBody;
+import com.palantir.logsafe.UnsafeArg;
+import java.io.IOException;
+import java.io.OutputStream;
+import org.apache.hc.core5.http.HttpEntity;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Adapts {@link HttpEntity} to a {@link RequestBody}.
+ */
+// Salt needs MIXED; other places use multipartBuilder.setType(MultipartBody.FORM);
+abstract class HttpEntityBodyRequestBodyAdapter implements RequestBody {
```

Comment:
I'm thinking we should do the same trick with FormUrlEncoded?

Reply:
Don't we already have a pure form-encoded body implementation in WC?

Reply:
yeah, that is defo a possibility. But thought might as well use the same trick for both of these. No preference though.